content/docs/reactivesearch/searchbase/overview/QuickStart.md

```diff
@@ -23,120 +23,222 @@ yarn add @appbaseio/searchbase
 ```
 
-##Basic Usage
-###Without suggestions
+## Basic Usage
 
-```js
-import SearchBase from '@appbaseio/searchbase';
+### A simple example
 
-// Instantiate the search base object
-const searchbase = new SearchBase({
-	index: 'gitxplore-latest-app',
-	credentials: 'LsxvulCKp:a500b460-73ff-4882-8d34-9df8064b3b38',
-	url: 'https://scalr.api.appbase.io',
-	dataField: ['name', 'description', 'name.raw', 'fullname', 'owner', 'topics'],
+The following example creates a single component of type `search` to render the results based on the input value.
+
+```js
+import { Component } from '@appbaseio/searchbase';
+
+// Instantiate the `Component`
+const searchComponent = new Component({
+	// Elasticsearch index name
+	index: 'gitxplore-app',
+	// Appbase credentials
+	credentials: 'a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61',
+	url: 'https://@arc-cluster-appbase-demo-6pjy6z.searchbase.io',
+	// Unique identifier for component
+	id: 'search-component',
+	// initialize with empty value
+	value: '',
+	// Database fields to perform the search
+	dataField: ['name', 'description', 'name.search', 'fullname', 'owner', 'topics'],
 });
 
+// Get the input element
 const searchElement = document.getElementById('search');
 
-// Bind the searchbase value to input value
-searchElement.value = searchbase.value;
-// Update the search input value in
+// Bind the searchComponent value to input value
+searchElement.value = searchComponent.value;
+
+// Update the search input value to searchComponent to fetch the results
 searchElement.addEventListener('input', e => {
-	searchbase.setValue(e.target.value);
+	// To fetch the suggestions based on the value changes
+	searchComponent.setValue(e.target.value);
 });
+
 // Build DOM when search results update
-searchbase.onResults = results => {
-	const resultsElement = document.getElementById('results');
-	resultsElement.innerHTML = '';
-	results.data.forEach(element => {
-		var node = document.createElement('LI'); // Create a <li> node
-		var resultNode = document.createTextNode(element.name); // Create a text node
-		node.appendChild(resultNode); // Append the text to <li>
-		resultsElement.appendChild(node);
-	});
-};
+searchComponent.subscribeToStateChanges(
+	change => {
+		const results = change.results.next;
+		const resultsElement = document.getElementById('results');
+		resultsElement.innerHTML = '';
+		results.data.forEach(element => {
+			var node = document.createElement('LI'); // Create a <li> node
+			var resultNode = document.createTextNode(element.name); // Create a text node
+			node.appendChild(resultNode); // Append the text to <li>
+			resultsElement.appendChild(node);
+		});
+	},
+	['results'],
+);
+
+// Fetch the default results at initial load
+searchComponent.triggerDefaultQuery();
 ```
 
 Add this in your HTML
 
-```js
+```html
 <input placeholder="type to search" id="search" />
 
 <div id="results"></div>
 ```
 
```

Comment:
Add a CSB link with the same example code in there so a user can visually see how this looks in the UI.